### ISSUE TITLE
MQTT connection weirdness

### ISSUE THREAD

**Issue by gilav** [edfungus/Crouton#34]
Hello

Had connection problem to the broker when not on localhost, but maybe is my fault... Investigate a newly installed crouton at this time, will update this note when I reach a conclusion (if I can)...

best regards  

Gilles

**Comment by djino44** [comment]
I have the same issue

**Comment by datenwissenschaften** [comment]
Same for me.

WebSocket connection to 'ws://[SANITIZED]:1883/mqtt' failed: Error during WebSocket handshake: net::ERR_CONNECTION_RESET

-----
Update
-----
This solved the problem:

http://blog.ithasu.org/2016/05/enabling-and-using-websockets-on-mosquitto/

**Comment by gilav** [comment]
Hello

The newly installed crouton works well. 
My problem was due to the fact that for unknown reason I had problem connecting to the WS port  (custom: 1084) of my mosquitto at home(router configured well), I changed the ports to have them around 5900 and now it is ok.
Also along my investigation I tried a more recent browserMqtt.js, which gave me other problems.... Maybe Edmund can explain why this .js seems to be the only library 'fixed' (i.e. not defined in package.json or something else. Sorry I am new to nodejs...)

I had a few css and/or visual glitch (on chrome browser):
- 3 checkbox text from MQTT broker setting almost unreadable (almost white on white background)> I commented the 'color: @fc-light;' in the app/framework/crouton-frame/crouton-frame.lass
-  on the crouton-chart-donut: I have a very small text label (class ct-label) that I would get rid of because it is too small, and I can have the centerSum which look better. Also the units label is too small.
Will try to improve the css for my usage.

Anyway this is a fantastic system, I can control my raspberry processes from my phone. Next step is to try the esp8266 stuff

gilles
